rapidfuzz/details/common.hpp

```diff
@@ -228,9 +228,9 @@ struct PatternMatchVector {
      * lookup key inside the hashmap using a similar collision resolution
      * strategy to CPython and Ruby
      */
-    uint64_t lookup(uint64_t key) const
+    int32_t lookup(uint64_t key) const
     {
-        uint64_t i = key % 128;
+        int32_t i = key % 128;
```

Comment:
The type of I has to stay uint64_t otherwise the following causes signed integer overflow:
```cpp
i = ((i * 5) + perturb + 1) % 128;
```
In this case I explicitly need the modulo behavior of unsigned integers (probably this is worth a comment)

Reply:
The `lookup` function returns the `i` and is called by 2 other functions, both of which use the returned value as indices, which can never be greater or equal to 128. How do you think if I cast `i` to an `uint64_t` within the modulo operation:
```
i = (static_cast<uint64_t>(i) * 5 + perturb + 1) % 128;
```